CMakeLists.txt

```
@@ -103,6 +107,43 @@ if("${_compiler_arch}" STREQUAL "")
     set(_compiler_arch ${CMAKE_SYSTEM_PROCESSOR})
 endif()
 
+# LLVM component setup
+message("Compiler architecture is ${_compiler_arch}")
+
+set(LLVM_COMPONENTS core demangle objcarcopts orcjit)
+
+if(${_compiler_arch} MATCHES "x64")
+    list(APPEND LLVM_COMPONENTS
+        x86asmparser
+        x86codegen
+        x86desc
+        x86info
+    )
+elseif(${_compiler_arch} MATCHES "aarch64")
+    list(APPEND LLVM_COMPONENTS
+        aarch64asmparser
+        aarch64codegen
+        aarch64desc
+        aarch64info
+    )
```

I ended up needing these changes to be able to compile on the M1:

```suggestion
elseif(${_compiler_arch} MATCHES "arm64")
    list(APPEND LLVM_COMPONENTS
        x86asmparser
        x86codegen
        armasmparser
        armcodegen
        aarch64asmparser
        aarch64codegen
        aarch64desc
        aarch64info
    )
```

---

This works for me too